packages/stack/proxy/src/proxy.js

```diff
@@ -210,6 +179,81 @@ export class Proxy {
+    clientSocket.on('close', () => currentReqManager.provider.disconnect());
+
+    // do the actual forward request to the node
+    currentReqManager.send(request, (error, result) => {
+      if (error) {
+        return cb(error);
+      }
+      // `result` contains our initial response from the node, ie
+      // subscription id. Any FUTURE data from the node that needs
+      // to be forwarded to the client connection will be handled
+      // in the `.on('data')` event below.
+      this.logger.debug(`Created subscription: ${result} for ${JSON.stringify(request.params)}`);
+      this.logger.debug(`Subscription request: ${JSON.stringify(request)} `);
+
+      // add the websocket req manager for this subscription to memory so it
+      // can be referenced later
+      this.nodeSubscriptions[result] = currentReqManager;
+
+      // send a response to the original requesting inbound client socket
+      // (ie the browser or embark) with the result of the subscription
+      // request from the node
+      response.result = result;
+      cb(null, response);
+    });
+
+    // Watch for `eth_subscribe` subscription data coming from the node.
+    // Send the subscription data back across the originating client
+    // connection.
+    currentReqManager.provider.on('data', async (result, deprecatedResult) => {
+      result = result || deprecatedResult;
+      this.logger.debug(`Subscription data received from node and forwarded to originating socket client connection: ${JSON.stringify(result)} `);
+
+      // allow modification of the node subscription data sent to the client
+      result = await this.emitActionsForResponse(result, result, clientSocket);
+      this.respondWs(clientSocket, result);
+    });
+  }
+
+  async handleUnsubscribe(request, response, cb) {
+    // kill our manually created long-living connection for eth_subscribe if we have one
+    const subscriptionId = request.params[0];
+    const currentReqManager = this.nodeSubscriptions[subscriptionId];
+
+    if (!currentReqManager) {
+      return this.logger.error(`Failed to unsubscribe from subscription '${subscriptionId}' because the proxy failed to find an active connection to the node.`);
+    }
+    // forward unsubscription request to the node
+    currentReqManager.send(request, (error, result) => {
+      if (error) {
+        return cb(error);
+      }
+      // `result` contains 'true' if the unsubscription request was successful
+      this.logger.debug(`Unsubscription result for subscription '${JSON.stringify(request.params)}': ${result} `);
+      this.logger.debug(`Unsubscription request: ${JSON.stringify(request)} `);
+
+
+      // if unsubscribe succeeded, disconnect connection and remove connection from memory
+      if (result === true) {
+        if (currentReqManager.provider && currentReqManager.provider.disconnect) {
+          currentReqManager.provider.disconnect();
+        }
+        delete this.nodeSubscriptions[subscriptionId];
+      }
```

Comment:
If result is `false`, should we do something about it? Does it mean it failed, or the connection was already closed? Should we delete the `nodeSubscription`?

Reply:
We could possibly log that information for the user to see (ie `Unsubsciption failed for subscription '${subscriptionId}'.`). But no, we should not delete the subscription as the subscription is not deleted and the socket should remain open at the point.